packages/upload-api/src/access/confirm.js

```diff
@@ -68,6 +68,12 @@ export async function confirm({ capability, invocation }, ctx) {
     // implement sudo access anyway.
     delegationProofs: delegationsResult.ok,
     expiration: Infinity,
+    attestationFacts: [
+      {
+        // include this so that the attestation CID is unique across confirmations
+        cause: invocation.cid,
+      }
+    ]
```

Comment:
Can you this just be a top level please ? And you could also include it in both delegations created

```suggestion
    // include this so that the attestation CID is unique across confirmations
    cause: invocation.cid,
```

Reply:
For the context we have another issue with the authorization (that this would allow us to resolve) where authorization succeeds before user clicks the email, that is because following function

https://github.com/web3-storage/w3up/blob/b232aa784c178cc6877982d7d9dfe4da06a06203/packages/access-client/src/agent-use-cases.js#L208C23-L213

uses not so great heuristics to determine when authorization has been completed

https://github.com/web3-storage/w3up/blob/b232aa784c178cc6877982d7d9dfe4da06a06203/packages/access-client/src/agent-use-cases.js#L97-L99


We could try to improve heuristics, but ideally we would have a way of just seeing if received delegation corresponds to the authorization invocation, which is why I'd like to capture these details somewhere.

Reply:
> ideally we would have a way of just seeing if received delegation corresponds to the authorization invocation,

huge +1. I was suggesting something similar to @travis the other day, but I didn't think to just use the CIDs of the precursor invocations as facts. love that idea